src/Symfony/Component/Form/ResolvedFormType.php

```diff
@@ -205,8 +205,16 @@ public function getOptionsResolver()
 
             $this->innerType->setDefaultOptions($this->optionsResolver);
 
+            if (method_exists($this->innerType, 'configureOptions')) {
```

Comment:
this ugly check should be remove. Instead, the default implementation of setDefaultOptions in the AbstractType should call the new method